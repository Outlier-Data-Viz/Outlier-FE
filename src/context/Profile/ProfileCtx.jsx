import { createContext, useContext, useMemo, useState } from "react";


const ProfileCtx = createContext();

function ProfileProvider({ children }) {
  const [loading, setLoading] = useState(true);
  const [session, setSession] = useState('')
  const [profile, setProfile] = useState({});

  // const value = useMemo(() => ({ profile, setProfile}), [profile]);
  return <ProfileCtx.Provider value={{ profile, setProfile, session, setSession }}>{children}</ProfileCtx.Provider>
};

const useProfile = () => {
  const ctx = useContext(ProfileCtx);

  if(ctx === undefined) {
    throw new Error('useProfile must be a child of Profile context');
  }

  return ctx;
};

export { ProfileCtx, ProfileProvider, useProfile};