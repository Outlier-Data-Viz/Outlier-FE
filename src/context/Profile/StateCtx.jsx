import { createContext, useContext, useState } from "react";

const StateCtx = createContext();

function StateProvider({ children }) {
  const [activeStates, setActiveStates] = useState([]);
  const [activeData, setActiveData] = useState([]);
  const [stateNames, setStateNames] = useState([]);

  return <StateCtx.Provider value={{stateNames, setStateNames, activeData, setActiveData, activeStates, setActiveStates}}>{children}</StateCtx.Provider>
}

const useActiveStates = () => {
  const ctx = useContext(StateCtx);

  if(ctx === undefined) {
    throw new Error('useActiveState hook must be a child of the State context provider');
  };

  return ctx;
};

export { StateCtx, StateProvider, useActiveStates};