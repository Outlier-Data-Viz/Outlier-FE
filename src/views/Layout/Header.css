:root {
  --pad: 1.2rem;
  --melon: #FFB5A7;
  --pale-pink: #FCD5CE;
  --isabelline: #F8EDEB;
  --peach: #F9DCC4;
  --peach-crayola: #FEC89A;
  --mauve: #86626E;
}

.header{
  padding: 20px;
  margin-bottom: 30px;
  display: flex;
  justify-content: space-between;
}
nav {
  display: flex;
  flex-direction: row;
  align-items: flex-start;
  gap: 10px;
  text-transform: uppercase;
}

.navigation a {
  color: var(--mauve);
  text-decoration: none;
  letter-spacing: 1px;
  font-size: 20px;
  padding: 10px;
}

.navLink {
  background-color: var(--peach);
}

.homeLink{ 
  background-color: var(--peach-crayola);
}
