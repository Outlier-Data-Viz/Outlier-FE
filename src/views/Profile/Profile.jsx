import { useNavigate } from "react-router-dom";
import { getProfile } from "../../services/profile";

function Profile() {
  const navigate = useNavigate();
  const { profile, setProfile } = useUser();
  const [loading, setLoading] = useState(true);

  useEffect(() => {
    try {
      const fetchProfile = async () => {
        const res = await getProfile();
        setProfile(res);
      }
      fetchProfile();
      if(!profile.username) {
        navigate('/create', {replace: true});
      } else 
        setLoading(false);
    } catch (err) {
        throw new Error('failed to fetch')
    }
  }, []);

  console.log(profile);

  return (
    loading ? <h1>Loading...</h1> :
    <>
      <section>
        <h1>{profile.email}</h1>
        <Link to='/edit'>Edit</Link>
      </section>
    </>
  );
};

export default Profile;