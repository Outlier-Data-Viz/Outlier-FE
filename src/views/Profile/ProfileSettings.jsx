import ProfileForm from "../../components/Forms/ProfileForm/ProfileForm";

function ProfileSettings() {
  return <ProfileForm />
};

export default ProfileSettings;