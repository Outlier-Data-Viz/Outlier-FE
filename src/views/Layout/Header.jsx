import { NavLink } from "react-router-dom";
import { useProfile } from "../../context/Profile/ProfileCtx";
import LoginHooks from "../../hooks/UseLogin";
import LogoutHooks from "../../hooks/UseLogout";

export default function Header() {
  const { profile } = useProfile();
  return (
    <header className='header'>
      <h1>Outlier</h1>
      {/* <h3>{profile.email}</h3> */}
      <NavLink to="/">Home</NavLink>
      <NavLink to="/instructions"> Instructions</NavLink>
      <NavLink to="/map"> Map</NavLink>
      <NavLink to="/data"> Data</NavLink>
      <NavLink to="/about"> About</NavLink>
      <NavLink to="/resources"> Resources</NavLink>
      <NavLink to="/profile"> Profile</NavLink>
      <LoginHooks />
      <LogoutHooks />
    </header>
  );
}