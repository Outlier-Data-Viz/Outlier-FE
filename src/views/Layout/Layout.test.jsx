import React from 'react';
import Layout from './Layout';
import { ProfileProvider } from '../../context/Profile/ProfileCtx';
import { BrowserRouter as Router } from 'react-router-dom';
import { render } from '@testing-library/react';

it('should render the layout', () => {
  const { container } = render(
    <Router>
      <ProfileProvider>
        <Layout />
      </ProfileProvider>
    </Router>
  );

  expect(container).toMatchSnapshot();
});
