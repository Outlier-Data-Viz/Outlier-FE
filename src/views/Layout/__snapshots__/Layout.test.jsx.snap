// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the layout 1`] = `
<div>
  <div>
    <header
      class="header"
    >
      <h1>
        Outlier
      </h1>
      <a
        aria-current="page"
        class="active"
        href="/"
      >
        Home
      </a>
      <a
        class=""
        href="/instructions"
      >
         Instructions
      </a>
      <a
        class=""
        href="/map"
      >
         Map
      </a>
      <a
        class=""
        href="/data"
      >
         Data
      </a>
      <a
        class=""
        href="/about"
      >
         About
      </a>
      <a
        class=""
        href="/resources"
      >
         Resources
      </a>
      <a
        class=""
        href="/profile"
      >
         Profile
      </a>
      <button
        class="authButton"
      >
        Auth w/ google
      </button>
      <button>
        Log Out
      </button>
    </header>
    <main />
    <footer>
      footer
    </footer>
  </div>
</div>
`;
